### DISCUSSION TITLE
what's the difference between invoke API(openAI API or Azure openAI API) directly and development a app by SK？

### DISCUSSION THREAD

**Discussion by lishengkai1984** [microsoft/semantic-kernel#1317] [Q&A]
what's the difference between invoke API(openAI API or Azure openAI API) directly and development a app by SK？What did SK do before and after invoke the API

**Comment by lukasz-appstream** [comment]
SK is not a client library for any specific api or service. Well, it does that too, but also much more.

It's an orchestrator framework that allows you to mix and match different semantic/ai tools (openAI API among them) in order to build an effective (and secure) yet modular and extendible AI application.
